pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -0,0 +1,273 @@
+		flags:            NoFix,
+	},
+	{
+		configKeySuffix:    "check-crc-nm-dispatcher-file",
+		checkDescription:   fmt.Sprintf("Checking if %s exists", crcNMDispatcherPath),
+		check:              checkCrcNMDispatcherFile,
+		fixDescription:     "Writing NetworkManager dispatcher file for crc",
+		fix:                fixCrcNMDispatcherFile,
+		cleanupDescription: fmt.Sprintf("Removing %s file", crcNMDispatcherPath),
+		cleanup:            removeCrcNMDispatcherFile,
+	},
+}
+
+func fixNMConfigFile(path string, content string, perms os.FileMode) error {
+	err := crcos.WriteToFileAsRoot(
+		fmt.Sprintf("write NetworkManager configuration to %s", path),
+		content,
+		path,
+		perms,
+	)
+	if err != nil {
+		return fmt.Errorf("Failed to write config file: %s: %v", path, err)
+	}
+
+	logging.Debug("Reloading NetworkManager")
+	sd := systemd.NewHostSystemdCommander()
+	if err := sd.Reload("NetworkManager"); err != nil {
+		return fmt.Errorf("Failed to restart NetworkManager: %v", err)
+	}
+
+	return nil
+}
+
+func removeNMConfigFile(path string) error {
+	if err := checkNetworkManagerInstalled(); err != nil {
+		// When NetworkManager is not installed, its config files won't exist
+		return nil
+	}
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		logging.Debugf("Removing NetworkManager configuration file: %s", path)
+		err := crcos.RemoveFileAsRoot(
+			fmt.Sprintf("removing NetworkManager configuration file in %s", path),
+			path,
+		)
+		if err != nil {
+			return fmt.Errorf("Failed to remove NetworkManager configuration file: %s: %v", path, err)
+		}
+
+		logging.Debug("Reloading NetworkManager")
+		sd := systemd.NewHostSystemdCommander()
+		if err := sd.Reload("NetworkManager"); err != nil {
+			return fmt.Errorf("Failed to restart NetworkManager: %v", err)
+		}
+	}
+	return nil
+}
+
+func checkCrcDnsmasqConfigFile() error {
+	logging.Debug("Checking dnsmasq configuration")
+	err := crcos.FileContentMatches(crcDnsmasqConfigPath, []byte(crcDnsmasqConfig))
+	if err != nil {
+		return err
+	}
+	logging.Debug("dnsmasq configuration is good")
+	return nil
+}
+
+func fixCrcDnsmasqConfigFile() error {
+	logging.Debug("Fixing dnsmasq configuration")
+	err := fixNMConfigFile(crcDnsmasqConfigPath, crcDnsmasqConfig, 0644)
+	if err != nil {
+		return err
+	}
+
+	logging.Debug("dnsmasq configuration fixed")
+	return nil
+}
+
+func removeCrcDnsmasqConfigFile() error {
+	return removeNMConfigFile(crcDnsmasqConfigPath)
+}
+
+func checkCrcNetworkManagerConfig() error {
+	logging.Debug("Checking NetworkManager configuration")
+	err := crcos.FileContentMatches(crcNetworkManagerConfigPath, []byte(crcNetworkManagerConfig))
+	if err != nil {
+		return err
+	}
+	logging.Debug("NetworkManager configuration is good")
+	return nil
+}
+
+func fixCrcNetworkManagerConfig() error {
+	logging.Debug("Fixing NetworkManager configuration")
+	err := fixNMConfigFile(crcNetworkManagerConfigPath, crcNetworkManagerConfig, 0644)
```

Comment:
Not a huge fan of abbr. for tech or names.
Esp. if the file contains both `NM` + `NetworkManager`

Note: purposely abbreviate text in this comment to indicate the confusing nature. It is inconsistent

Reply:
Fixed